test-status/api.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright 2019, the AMP HTML authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const {getCheckRunId, getPullRequestSnapshot} = require('./db');
+
+/**
+ * Create a parameters object for a new status check line.
+ *
+ * @param {string} pullRequestSnapshot pull request snapshot from database.
+ * @param {string} type tests type slug.
+ * @param {string} status status action.
+ * @return {!Object} a parameters object for github.checks.create|update.
+ */
+function createNewCheckParams(pullRequestSnapshot, type, status) {
+  const params = {
+    owner: pullRequestSnapshot.owner,
+    repo: pullRequestSnapshot.repo,
+    name: `ampproject/tests/${type}`,
+    head_sha: pullRequestSnapshot.head_sha,
+  };
+  switch (status) {
+    case 'queued':
+      Object.assign(params, {
+        status: 'queued',
+        output: {
+          title: `${type} tests are queued on Travis`,
+          summary: `The ${type} tests are queued to run on Travis. Watch ` +
+            'this space for results in a few minutes!',
```

Comment:
I think most of this message will be truncated. Perhaps shorten to something like: `${type} tests queued on Travis. Waiting for status...`?

![image](https://user-images.githubusercontent.com/26553114/54136463-3f05c480-43f2-11e9-916f-e6b4a9fc80b4.png)

Reply:
The title is the short text that appears in the screenshot about - the summary is what's hidden behind the "details" link. Even if it gets truncated, I'm confident the first 4 works (`${type} tests are queued…`) will be visible